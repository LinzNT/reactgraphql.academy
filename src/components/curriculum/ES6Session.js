import React from 'react'
import Ul, { Li } from '../layout/Ul'
import Session from './Session'

const ES6Session = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>
        Understanding language updates - difference between ES6, ES7, and ESNEXT
      </Li>
      <Li>Arrow Functions, Class syntax, Template strings</Li>
      <Li>Destructuring</Li>
      <Li>Default parameters, Rest operator, Spread operator</Li>
      <Li>Let and Const vs Var</Li>
      <Li>Rest + Spread properties</Li>
      <Li>ES6 iterators and functional programming in JS</Li>
      <Li>Modules</Li>
      <Li>Promises</Li>
      <Li>ES6 exercise using TDD</Li>
    </Ul>
  </Session>
)

export default ES6Session
