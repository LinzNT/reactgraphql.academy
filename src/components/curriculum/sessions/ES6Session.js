import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

const ES6Session = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>Modern JavaScript for React - ES6 and beyond</Li>
      <Li>Arrow Functions, Class syntax, Template strings</Li>
      <Li>Destructuring</Li>
      <Li>Default parameters, REST operator, Spread operator</Li>
      <Li>Let and Const vs Var</Li>
      <Li>REST + Spread properties</Li>
      <Li>Iterators and functional programming in JS</Li>
      <Li>Modules</Li>
      <Li>Promises</Li>
      <Li>ES6 exercise using TDD</Li>
    </Ul>
  </Session>
)

export default ES6Session
