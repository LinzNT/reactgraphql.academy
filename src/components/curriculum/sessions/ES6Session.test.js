import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import ES6Session from './ES6Session'

describe('<ES6Session />', () => {
  it('should render component', () => {
    const wrapper = shallow(<ES6Session />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
