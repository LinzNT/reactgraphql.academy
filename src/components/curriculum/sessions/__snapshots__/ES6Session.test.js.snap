// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ES6Session /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      Modern JavaScript for React - ES6 and beyond
    </styled.li>
    <styled.li>
      Arrow Functions, Class syntax, Template strings
    </styled.li>
    <styled.li>
      Destructuring
    </styled.li>
    <styled.li>
      Default parameters, Rest operator, Spread operator
    </styled.li>
    <styled.li>
      Let and Const vs Var
    </styled.li>
    <styled.li>
      Rest + Spread properties
    </styled.li>
    <styled.li>
      ES6 iterators and functional programming in JS
    </styled.li>
    <styled.li>
      Modules
    </styled.li>
    <styled.li>
      Promises
    </styled.li>
    <styled.li>
      ES6 exercise using TDD
    </styled.li>
  </styled.ul>
</Session>
`;
